lib/MusicBrainz/Server/Controller/Role/Tag.pm

```diff
@@ -68,6 +68,10 @@ sub _vote_on_tags {
         $c->detach;
     }
 
+    if (!$c->user->has_confirmed_email_address) {
+        $c->detach('/error_401');
+    }
+
```

Comment:
That returns an HTML page, right? Since this is a JSON endpoint we should probably add `|| !$c->user->has_confirmed_email_address` to the condition above instead.